src/msg/async/dpdk/DPDKStack.cc

```diff
@@ -250,8 +250,26 @@ void DPDKStack::spawn_worker(unsigned i, std::function<void ()> &&func)
   // if dpdk::eal::init already called by NVMEDevice, we will select 1..n
   // cores
   assert(rte_lcore_count() >= i + 1);
+  auto coremask_arg = cct->_conf->get_val<std::string>("ms_dpdk_coremask");
+  int first_slave_coreid = -1;
+  try {
+    auto core_value = stoull(coremask_arg, nullptr, 16);
+    first_slave_coreid = ffsll(core_value) + 1;
+  } catch (const std::logic_error& e) {
+    lderr(cct) << __func__ << " invalid ms_spdk_coremask: "
+	 << coremask_arg << dendl;
+    ceph_abort();
+  }
+
+  if (first_slave_coreid == 1)
+  {
+    lderr(cct) << __func__ << " invalid ms_spdk_coremask: "
+	 << at least one core is needed << dendl;
```

Comment:
this does not compile.

Reply:
thx, this log infor can be changed just like this:
  if (first_slave_coreid == 1)
  {
    lderr(cct) << __func__ << " invalid ms_dpdk_coremask: "
	 << at least one core is needed << dendl;

Reply:
we should allocate cores for each process of ceph， when running with dpdk. So we have to make clear the core of each process. If a process has only a primary core and no auxiliary core, then this process can not start success. Based on this consideration, I added this logic .